### PR TITLE
feat(analytics): adds basic signup stream

### PR DESCRIPTION
**Pull request by andrewxhill**
Signed-off-by: Andrew Hill <andrew@textile.io>

### CODE REVIEW

File: cmd/hubd/main.go [hunk 1]
```diff
@@ -96,6 +96,10 @@ var (
 				Key:      "dns.token",
 				DefValue: "",
 			},
+			"segmentKey": {
```
Comment (sanderpick) [inline cmd/hubd/main.go:99]: This should be on the billing daemon

File: api/billingd/client/client.go [hunk 2]
```diff
@@ -54,6 +67,19 @@ func (c *Client) CreateCustomer(ctx context.Context, key thread.PubKey, opts ...
 	if err != nil {
 		return "", err
 	}
+
+	// Create new analytics entry
```
Comment (sanderpick) [inline api/billingd/client/client.go:71]: Shouldn't this happen in the service handler, not the client?

File: api/billingd/client/client.go [hunk 3]
```diff
@@ -54,6 +67,19 @@ func (c *Client) CreateCustomer(ctx context.Context, key thread.PubKey, opts ...
...
+	if c.sc != nil {
+		c.sc.Enqueue(analytics.Identify{
+			UserId: key.String(),
+			Traits: analytics.NewTraits().
+				SetEmail(args.email).
+				Set("parent_key", parentKey).
+				Set("account_type", args.accountType).
```
Comment (sanderpick) [inline api/billingd/client/client.go:78]: Nice, maybe we can put `account_type` on the `Customer` object. Good info.

File: core/usage_interceptor.go [hunk 4]
```diff
@@ -56,6 +56,7 @@ func (t *Textile) preUsageFunc(ctx context.Context, method string) (context.Cont
 	if t.bc == nil {
 		return ctx, nil
 	}
+
```
Comment (sanderpick) [inline core/usage_interceptor.go:59]: Adding newlines?

**Review by sanderpick — COMMENTED** [review]
Will be cool to these data come in! Left some comments.

Related to the general approach, instead of collection each event manually, i.e., "customer created", we could just pipe all the [stripe events](https://stripe.com/docs/api/events/types) into segment using the [webhook handler](https://github.com/textileio/textile/blob/master/api/billingd/gateway/gateway.go#L120). Maybe that's what this [stripe integration](https://segment.com/docs/connections/sources/catalog/cloud-apps/stripe/) does.

File: api/billingd/service/service.go [hunk 5]
```diff
@@ -362,15 +376,16 @@ func (s *Service) Stop(force bool) error {
...
-	Status     string `bson:"status"`
-	Balance    int64  `bson:"balance"`
-	Billable   bool   `bson:"billable"`
-	Delinquent bool   `bson:"delinquent"`
-	CreatedAt  int64  `bson:"created_at"`
+	Key         string `bson:"_id"`
+	CustomerID  string `bson:"customer_id"`
+	ParentKey   string `bson:"parent_key"`
+	Email       string `bson:"email"`
+	AccountType string `bson:"account_type"`
```
Comment (sanderpick) [inline api/billingd/service/service.go:383]: How about using the [account type](https://github.com/textileio/textile/blob/master/mongodb/accounts.go#L45) directly?

Comment (andrewxhill) [reply api/billingd/service/service.go:383]: what does that look like in the request object? just convert it to int64 and then back?

Comment (sanderpick) [reply api/billingd/service/service.go:383]: I think so, yep. Your new client option could take `mdb.AccountType`, the just wrap that it `int64()` in the client method, the when creating the customer, `mdb.AccountType(req.AccountType)`

Comment (andrewxhill) [reply api/billingd/service/service.go]: k, done + rebased

**Review by sanderpick — APPROVED** [review]
LGTM!

**Pull request merged**